pkg/gitops/filter.go

```diff
@@ -0,0 +1,90 @@
+package gitops
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+)
+
+func (cr *CustomRepo) FilterCommits(author string, since time.Time, until time.Time, resource string, namespace string, name string) ([]object.Commit, error) {
+	var logopts git.LogOptions
+	var filteredCommits []object.Commit
+
+	cr.Mutex.Lock()
+	defer cr.Mutex.Unlock()
+	ref, err := cr.Repo.Head()
+	if err != nil {
+		return filteredCommits, fmt.Errorf("could not get ref head from repository: %w", err)
+	}
+
+	logopts.From = ref.Hash()
+	if !since.IsZero() {
+		logopts.Since = &since
+	}
+	if !until.IsZero() {
+		logopts.Until = &until
+	}
+	var resources []string
+	if resource != "" && namespace == "" && name != "" {
+		return filteredCommits, fmt.Errorf("Error (FilterCommits): cannot provide a resource and a name without a namespace")
+	} else if resource == "" && namespace != "" {
+		resources = []string{"k8s-policies", "antrea-policies", "antrea-cluster-policies", "antrea-tiers"}
+	} else {
+		resources = []string{resource}
+	}
+	for _, r := range resources {
+		setPathFilter(r, namespace, name, &logopts)
+		tempCommits, err := cr.filter(&logopts, func(commit *object.Commit) bool {
```

Comment:
I don't understand why this isn't:
```go
var err error
filteredCommits, err = cr.filter(...)
```
it really doesn't seem necessary to make a copy `tempCommits` -> `filteredCommits`

Reply:
It only exists to append to filteredCommits, where filteredCommits holds a list of all the commits, but tempCommits is just a list of commits for the current resource.

Reply:
I see that makes sense. I read too fast. This is good for now.